STL_Extension/include/CGAL/Handle_for.h

```diff
@@ -63,7 +81,11 @@ class Handle_for
     {
         pointer p = allocator.allocate(1);
         new (&(p->t)) element_type(); // we get the warning here
+#ifdef CGAL_HANDLE_FOR_USE_BOOST_ATOMIC_COUNTER
+        new (&(p->count)) boost::detail::atomic_count(1);
+#else // not Boost atomic counter
         p->count = 1;
+#endif // not Boost atomic counter
```

Comment:
Yes, it should. Well, the constructor would not call placement new, just initialize its members. `template<class...T>RefCounted(T&&...t):t(std::forward<T>(t)...), count(1){}`. Used as `new(p)RefCounted(t);`.